lib/new_relic/agent/log_event_aggregator.rb

```diff
@@ -90,6 +90,42 @@ def record(formatted_message, severity)
         nil
       end
 
+      def record_json(log)
+        return unless enabled?
+
+        severity = log['level'] || 'UNKNOWN'
+
+        if NewRelic::Agent.config[METRICS_ENABLED_KEY]
+          @counter_lock.synchronize do
+            @seen += 1
+            @seen_by_severity[severity] += 1
+          end
+        end
+
+        return if severity_too_low?(severity)
+        # The message key only exists if users manually add log statements
+        return if log.key?(:message) && (log.key?(:message).nil? || llog.key?(:message).empty?)
+        return unless NewRelic::Agent.config[FORWARDING_ENABLED_KEY]
+        return if @high_security
+
+        txn = NewRelic::Agent::Transaction.tl_current
+        priority = LogPriority.priority_for(txn)
+        message = get_json_message(log)
```

Comment:
I think given the earlier check for `log[:message].empty?`, you can just do:

```suggestion
        message = log[:message]
```

and then remove the `get_json_message` method.

Reply:
We still need to send an empty string in place of the message per spec. Since many logstasher logs won't have a message, doing `message = log['message']` [now](https://github.com/newrelic/newrelic-ruby-agent/pull/2559/files#diff-680b1e6a8547301e66b77f383e273ef141166e5986c9a1180625081dc8e271bcR113)

Reply:
My concern is that you can't reach this line with a `nil` message because of the early return you perform earlier when seeing a `nil` message. If `nil` is ok and needs to be converted to `''` to save space over a JSON `null`, that's cool - but I think you need to then drop the early return logic that prevents `nil`.

https://github.com/newrelic/newrelic-ruby-agent/pull/2559/files#diff-680b1e6a8547301e66b77f383e273ef141166e5986c9a1180625081dc8e271bcR98

Reply:
I was originally attempting to keep it in line with Logger instrumentation [logic](https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/log_event_aggregator.rb#L73), but in LogStasher's case, we first check if there even is a `message` attribute because oftentimes for LogStasher there isn't. 

I suppose this check would only be hit if for some reason a user manually logged something with a `nil` or `''` message, which maybe should be allowed if that is their intent. Thoughts on removing this line entirely?

Reply:
By 'this check' and the one I'm thinking to remove, I mean [this](https://github.com/newrelic/newrelic-ruby-agent/pull/2559/files#diff-680b1e6a8547301e66b77f383e273ef141166e5986c9a1180625081dc8e271bcR98)

Reply:
Ah, my fault for misreading things. Sorry, @hannahramadan. I was commenting as if you were returning early by just evaluating `(log['message'].nil? || log['message'].empty?)`, but you're not - you're only applying that check to payloads that have a `'message'` key in the first place with your `if log.key?('message')` check that I skipped over / forgot about.

No changes necessary / great as-is.